backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/validator/AnsibleRunnerCleanUpService.java

```diff
@@ -0,0 +1,78 @@
+package org.ovirt.engine.core.bll.validator;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.concurrent.ManagedScheduledExecutorService;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.ovirt.engine.core.common.BackendService;
+import org.ovirt.engine.core.common.config.Config;
+import org.ovirt.engine.core.common.config.ConfigValues;
+import org.ovirt.engine.core.common.utils.ansible.AnsibleConstants;
+import org.ovirt.engine.core.dal.dbbroker.auditloghandling.AuditLogDirector;
+import org.ovirt.engine.core.utils.threadpool.ThreadPools;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class AnsibleRunnerCleanUpService implements BackendService {
+
+    private static Logger log = LoggerFactory.getLogger(AnsibleRunnerCleanUpService.class);
+
+    @Inject
+    private AuditLogDirector auditLogDirector;
+
+    @Inject
+    @ThreadPools(ThreadPools.ThreadPoolType.EngineScheduledThreadPool)
+    private ManagedScheduledExecutorService executor;
+
+    @PostConstruct
+    public void scheduleJob() {
+        double interval = Config.<Double> getValue(ConfigValues.AnsibleRunnerArtifactsCleanupCheckTimeInHours);
+        final int HOURS_TO_MINUTES = 60;
+        long intervalInMinutes = Math.round(interval * HOURS_TO_MINUTES);
+
+        executor.schedule(this::checkExecutionTimeStamp,
+                intervalInMinutes,
+                TimeUnit.MINUTES);
+    }
+
+    private void checkExecutionTimeStamp() {
+        int artifactsLifeTime = Config.<Integer> getValue(ConfigValues.AnsibleRunnerArtifactsLifetimeInDays);
+        Stream.of(new File(AnsibleConstants.ANSIBLE_RUNNER_PATH.toString()).listFiles()).forEach(file -> {
+            long creationInDays = 0;
+            try {
+                creationInDays = Files.readAttributes(file.toPath(), BasicFileAttributes.class)
+                        .creationTime()
+                        .to(TimeUnit.DAYS);
+            } catch (IOException e) {
+                log.error("Failed to read file attributes: {}", file.toPath());
+                log.debug("Exception: ", e);
+            }
+            long todayInDays = FileTime.fromMillis(new Date().getTime()).to(TimeUnit.DAYS);
+            if (todayInDays - creationInDays > artifactsLifeTime) {
```

Comment:
so it line 57 raises and we catch and log it, creationInDays is 0 and since todayInDays - 0 is always > artifactsLifeTime we wipe out all artifacts...

Reply:
right! sorry, fixing.